### ISSUE TITLE
Edit source code in text editor without tags

### ISSUE THREAD

**Issue by stemod** [sulu/sulu#3191]
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | 1.3.0
| Browser Version | Chrome v56

#### Actual Behavior

How does Sulu behave at the moment? 
Text editor does not load if tags are missing.

#### Expected Behavior

What is the behavior you expect?
Tags should be added automatically.

#### Steps to Reproduce

What are the steps to reproduce this bug? Please add code examples,
screenshots or links to GitHub repositories that reproduce the problem.
Edit text in text editor (source) and save without tags.

**Comment by alexander-schranz** [comment]
@stemod is the data come from an import or how does this happens that the tags are lost?

The ckeditor have different enter mode so in your case you maybe should switch the editor to br mode where no tags are needed:

e.g.:

```
        <property name="article" type="text_editor">
            <meta>
                <title lang="en">Article</title>
                <title lang="de">Artikel</title>
            </meta>
            <params>
                 <param name="enter_mode" value="br" /> <!-- note sure if enter-mode, enterMode or enter_mode -->
            </params>
        </property>
```

**Comment by wachterjohannes** [comment]
@stemod please fill in all the sections and remove the default text there.

Can you please add also any javascript errors which you can find in the console - while rendering a page?

**Comment by danrot** [comment]
@stemod You should really add more information the next time 😉 

The problem only appears if you switch the ckeditor to code mode as in the gif of @alexander-schranz. If you do not return the editor to normal mode before saving the tags will not be added...

**Comment by oafx** [comment]
maybe worth mentioning that i stumbled over this while using data fixtures.. if you load `text_editor` fixtures without html tags sulu kind of crashes (endless loading indicator in the top bar). I know there is an error being thrown in the javascript console in these cases but it'd be great if there was another kind of visual feedback if something broke instead of just displaying the loading indicator..

**Comment by danrot** [comment]
@oafx Yeah, you are completely right. The question is how we fix that: Should we try to just wrap the data in a HTML tag, so that editor doesn't crash? Or should there be some different kind of feedback, as you've mentioned...

**Comment by wachterjohannes** [comment]
only an idea - without testing it: maybe we could check on save the state of the editor and if the editor is in code-mode trigger the mode change. @danrot @alexander-schranz what did you think?

**Comment by danrot** [comment]
Doesn't solve the problem from @oafx, since he has already inserted it wrong in the fixtures.

**Comment by wachterjohannes** [comment]
ah, your right I overread this.

**Comment by oafx** [comment]
i don't know if it is nessecary to catch faulty data that is insertet via fixtures.. i mean loading fixtures is certainly not something a user should be doing without assistance/supervision of a dev.. so i wouldn't make this a priority if you ask me.. 

but: while sulu seems to me very stable, and i can't think of another way right now to break the admin interface, it'd still be great to have in general some kind of error modal or something else if it does break.. right now you just have the endless loading indicator and if you press save nothing happens.. it's especially frustrating for customers since they won't notice something is off/wrong.

**Comment by danrot** [comment]
Well, but it can still cost the developer a lot of time (I assume it was the same thing for you). So maybe we should validate the value being written in the content type on the server side. Then it's also not possible to insert wrong values via a fixture. And we should probably switch the text editor back to text mode before saving, if it has been in code mode.

**Comment by danrot** [comment]
fixed in #3687
